ksql-engine/src/test/java/io/confluent/ksql/QueryTranslationTest.java

```diff
@@ -104,23 +105,50 @@ public void shouldBuildAndExecuteQueries() {
   }
 
   @Parameterized.Parameters(name = "{0}")
-  public static Collection<Object[]> data() throws IOException {
-    final String topologyVersion = System.getProperty(TOPOLOGY_VERSION_PROP, CURRENT_TOPOLOGY_VERSION);
-    final String topologyDirectory = TOPOLOGY_CHECKS_DIR + "/" + topologyVersion;
-    final Map<String, TopologyAndConfigs> expectedTopologies = loadExpectedTopologies(topologyDirectory);
+  public static Collection<Object[]> data() {
+    final String[] topologyVersions =
+        System.getProperty(TOPOLOGY_VERSIONS_PROP, CURRENT_TOPOLOGY_VERSIONS)
+            .split(TOPOLOGY_VERSIONS_DELIMITER);
+    final List<TopologiesAndVersion> expectedTopologies = loadTopologiesAndVersions(topologyVersions);
     return buildTestCases()
-          .peek(q -> {
-            final TopologyAndConfigs topologyAndConfigs = expectedTopologies.get(formatQueryName(q.getName()));
-            // could be null if the testCase has expected errors, no topology or configs saved
-            if (topologyAndConfigs !=null) {
-              q.setExpectedTopology(topologyAndConfigs.topology);
-              q.setPersistedProperties(topologyAndConfigs.configs);
-            }
-          })
+          .flatMap(q -> buildVersionedTestCases(q, expectedTopologies))
           .map(testCase -> new Object[]{testCase.getName(), testCase})
           .collect(Collectors.toCollection(ArrayList::new));
   }
 
+  private static List<TopologiesAndVersion> loadTopologiesAndVersions(final String[] topologyVersions) {
+    return Stream.of(topologyVersions)
+        .map(topologyVersion -> {
+          final String topologyDirectory = TOPOLOGY_CHECKS_DIR + "/" + topologyVersion;
+          try {
+            return new TopologiesAndVersion(topologyVersion, loadExpectedTopologies(topologyDirectory));
```

Comment:
Not really part of your change, but if we changed `findExpectedTopologyFiles` to catch-and-throw IOException in a RuntimeException, you could get rid of this try/catch/throw here, which would be cleaner:

```
return Stream.of(topologyVersions)
    .map(v -> new TopologiesAndVersion(v, loadExpectedTopologies(TOPOLOGY_CHECKS_DIR + v)))
    .collect(...)
```

Reply:
Done.